src/App.tsx

```diff
@@ -1,61 +1,114 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { todo } from './components/types';
 
 export const App = () => {
+  const editedTodos = todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId) || {
+      id: 0,
+      name: 'Unknown User',
+      username: 'unknown',
+      email: 'unknown@example.com',
+    },
+  }));
+  
+  const [todos, setTodos] = useState<todo[]>(editedTodos)
+  const [title, setTitle] = useState('')
+  const [userId, setUserId] = useState(0)
+  const [titleError, setTitleError] = useState(false)
+  const [userError, setUserError] = useState(false)
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    let isError = false;
+
+    if (title.trim() === '') {
+      setTitleError(true);
+      isError = true;
+    }
+    if (userId === 0) {
+      setUserError(true);
+      isError = true;
+    }
+
+    if( isError) {
+      return;
+    }
+
+    const newTodo: todo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      title: title,
+      completed: false,
+      userId: userId,
+      user:   usersFromServer.find(user => user.id === userId) || {
```

Comment:
The logic to find the user by `userId` is correct, but ensure that `usersFromServer` is always available and populated to avoid potential runtime errors.